library/std/src/thread/scoped.rs

```diff
@@ -107,6 +107,24 @@ impl ScopeData {
 /// a.push(4);
 /// assert_eq!(x, a.len());
 /// ```
+///
+/// # Lifetimes
+///
+/// Scoped threads involve two lifetimes: `'scope` and `'env`.
+///
+/// The `'scope` lifetime represents the lifetime of the scope itself.
+/// That is: the time during which new scoped threads may be spawned,
+/// and also the time during which they might still be running.
+/// Once this lifetime ends, all scoped threads are joined.
```

Comment:
I care about this one because that's the right ordering: `'scope` by very definition, outlives the `Scope` and `Packet` instances, so the right property is rather "once the scoped threads are all joined, this lifetime may end", hence my suggestion